pkg/receive/handler.go

```diff
@@ -271,17 +274,27 @@ func (h *Handler) testReady(f http.HandlerFunc) http.HandlerFunc {
 	}
 }
 
-func (h *Handler) getStats(r *http.Request, statsByLabelName string) (*tsdb.Stats, error) {
+func (h *Handler) getStats(r *http.Request, statsByLabelName string) ([]statusapi.TenantStats, *api.ApiError) {
 	if !h.isReady() {
-		return nil, fmt.Errorf("service unavailable")
+		return nil, &api.ApiError{Typ: api.ErrorInternal, Err: fmt.Errorf("service unavailable")}
 	}
 
 	tenantID := r.Header.Get(h.options.TenantHeader)
+	getAllTenantStats := r.FormValue(AllTenantsQueryParam) == "true"
+	if getAllTenantStats && tenantID != "" {
+		err := fmt.Errorf("using both the %s parameter and the %s header is not supported", AllTenantsQueryParam, h.options.TenantHeader)
+		return nil, &api.ApiError{Typ: api.ErrorBadData, Err: err}
+	}
+
+	if getAllTenantStats {
+		return h.options.TSDBStats.TenantStats(statsByLabelName), nil
+	}
+
 	if tenantID == "" {
 		tenantID = h.options.DefaultTenantID
 	}
 
-	return h.options.GetTSDBStats(tenantID, statsByLabelName), nil
+	return h.options.TSDBStats.TenantStats(statsByLabelName, tenantID), nil
```

Comment:
There is still no way to query multiple tenants (not all) from the API. Do we want to provide this capability to users?

Reply:
My thought was that getting stats for all tenants should be good enough for most users that want to identify where high cardinality series come from. But we can add it if you think the use case is going to be common.

Reply:
It would be great if we can make it more extensible to support those use cases at this stage. If you think the current implementation is good to be extended for those cases in the future then I am good.

Reply:
Yes, I think we should be good to extend functionality in the future. We can pass multiple tenant IDs [here](https://github.com/thanos-io/thanos/pull/5470/files#diff-61eda56d07cd51eb9fde89c6bd9f1fe7e260edb05eb8fb4c1c81aa258f034331R393
). So the main question will be how the users will provide the tenant IDs, whether as comma-separated values in the `THANOS-TENANT` header or something else.